### PR TITLE
App state refactor & menu drawer implementation

### PR DESCRIPTION
**Pull request by j-nordin**
This PR is a refactor och the applications states, navController and scaffold. 

An implementation of a menu drawer is also done, not yet linked to any views.

Some refactoring for card pages is also done to support the detailed page wrapper and menu drawer.

### CODE REVIEW

File: androidApp/src/main/java/com/EENX15_22_17/digital_journal/android/screens/arrival/patientinfo/ArrivalScreen.kt [hunk 1]
```diff
@@ -9,13 +9,16 @@ import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
```
Comment (axelhertzberg) [inline androidApp/src/main/java/com/EENX15_22_17/digital_journal/android/screens/arrival/patientinfo/ArrivalScreen.kt:9]: This file will be changed in branch arrivalRefactor. Good for now

**Review by axelhertzberg — APPROVED** [review]
Everything is working goog.
Important note is that ArrivalScreen doesn't look like it should. This will be fixed with titledsection in arrivalRefactor branch.

File: androidApp/src/main/java/com/EENX15_22_17/digital_journal/android/AppNavigation.kt [hunk 2]
```diff
@@ -21,7 +21,7 @@ import com.EENX15_22_17.digital_journal.android.screens.triage.healthHistory.Hea
 import com.EENX15_22_17.digital_journal.android.screens.triage.previousCare.TempPreviusCare
 
 sealed class Screen(val route: String) {
-    object Overview : Screen(route = "overview")
+    object Current : Screen(route = "current")
```
Comment (rickardlantz) [inline androidApp/src/main/java/com/EENX15_22_17/digital_journal/android/AppNavigation.kt:24]: I think this name is a bit vague (not the the previous name was much better though).

In general I think we should decide on better naming for the different views, allowing for longer ones if they help with intelligibility. This should be discussed in a group though, so it's not something that will affect this PR.

**Review by rickardlantz — APPROVED** [review]
This looks really good code-wise, and the drawer UI part is very close to Figma!

Good usage of TODO:s as well :)

**Pull request merged**